### DISCUSSION TITLE
Different approach to Ingedients.

### DISCUSSION THREAD

**Discussion by christianreiss** [mealie-recipes/mealie#959] [Feature Request]
Hey,

as you experienced, doing the ingredients the way they are are a pain in the ... especially for shopping lists.

1x  1/2 spoon Salt
1x 1 spoon Salt
etc

Also, If you have a recipe for 4 burgers using 275 grams of ground Meat, well get your calculator our for 3 burgers. 

Here is a better approach: handle ingredients like tags, [carrot], [onion], [ground meat]. When you create a new recipe you select the ingredients out of the pool if ingredients, easily able to add new ones. Each ingredient in a recipe has two values: a) what and b) amount, so

[4] [carrots] and [2] onions for this recipe, another meal has
[2] [carrots] and [400] [ground meat]. This now easily allows to...

a) group all meals of the week for the shopping list [carrots, onions and ground meat] and simply add their values together.
b) Use a slider/scale to increase the portions from 4 burgers to 3 or 8.

Waaay down the road you could edit like [sugar] having 4 kcal per gram. If your meal uses 10 grams of sugar you can now add up the containing kcal and amounts.

Thanks for your consideration!
-Chris.

**Comment by Sennevds** [comment]
Checkout the beta version: https://beta.mealie.io/ the ingredient part is completely rewritten including a parser for the amount and ingredient

**Comment by christianreiss** [comment]
That was a) fast and b) wow.
